server/isomorphic-routing/index.js

```diff
@@ -6,22 +6,32 @@ import { setSection as setSectionMiddlewareFactory } from '../../client/controll
 
 export function serverRouter( expressApp, setUpRoute, section ) {
 	return function( route, ...middlewares ) {
-		expressApp.get(
-			route,
-			setUpRoute,
-			combineMiddlewares(
-				setSectionMiddlewareFactory( section ),
-				...middlewares
-			),
-			serverRender
-		);
+		if ( middlewares.length === 0 && typeof route === 'function' && route.length === 3 ) {
+			// No route def -- the route arg is really an error-handling middleware
+			expressApp.use( ( err, req, res, next ) => {
+				route( err, req.context, next );
+			} );
+			expressApp.use( ( err, req, res, next ) => { // eslint-disable-line no-unused-vars
```

Comment:
I think Express uses the arity of the middleware to determine if it's an error handler (n=4) or a regular mw (n=3)...